pkg/cache/cache_test.go

```diff
@@ -271,6 +280,35 @@ func CacheTest(createCacheFunc func(config *rest.Config, opts cache.Options) (ca
 					Expect(out).NotTo(Equal(knownPod2))
 				})
 
+				It("should not deep copy the object if UnsafeDisableCacheDeepCopy is enabled", func() {
+					if !opts.UnsafeDisableCacheDeepCopy {
+						Skip("skip for UnsafeDisableCacheDeepCopy disabled")
+					}
+					By("getting a specific pod from the cache twice")
+					podKey := client.ObjectKey{Name: "test-pod-2", Namespace: testNamespaceTwo}
+					out1 := &corev1.Pod{}
+					Expect(informerCache.Get(context.Background(), podKey, out1)).To(Succeed())
+					out2 := &corev1.Pod{}
+					Expect(informerCache.Get(context.Background(), podKey, out2)).To(Succeed())
+
+					By("verifying the pods have the same pointer addresses")
+					Expect(fmt.Sprintf("%p", out1)).NotTo(Equal(fmt.Sprintf("%p", out2)))
```

Comment:
Nit: here and elsewhere use [`BeIdenticalTo`](https://onsi.github.io/gomega/#beidenticaltoexpected-interface) instead of string comparisons of pointer values.

Reply:
done